uportal-war/src/main/java/org/jasig/portal/groups/pags/dao/EntityPersonAttributesGroupStore.java

```diff
@@ -354,20 +350,37 @@ private IPersonTester initializeTester(IPersonAttributesGroupTestDefinition test
         }
    }
 
-    private Set<IEntityGroup> getContainingGroups(String name, Set<IEntityGroup> groups) throws GroupsException
-    {
+    private Set<IEntityGroup> getContainingGroups(String name, Set<IEntityGroup> groups) throws GroupsException {
         logger.debug("Looking up containing groups for {}", name);
-        Set<IPersonAttributesGroupDefinition> pagsGroups = personAttributesGroupDefinitionDao.getPersonAttributesGroupDefinitionByName(name);
-        IPersonAttributesGroupDefinition pagsGroup = pagsGroups.iterator().next();
+        IPersonAttributesGroupDefinition pagsGroup = getPagsGroupDefByName(name);
         Set<IPersonAttributesGroupDefinition> pagsParentGroups = personAttributesGroupDefinitionDao.getParentPersonAttributesGroupDefinitions(pagsGroup);
-        for (IPersonAttributesGroupDefinition parent : pagsParentGroups) {
+        for (IPersonAttributesGroupDefinition pagsParent : pagsParentGroups) {
+            IEntityGroup parent = convertPagsGroupToEntity(pagsParent);
             if (!groups.contains(parent)) {
-                groups.add(convertPagsGroupToEntity(parent));
-                getContainingGroups(parent.getName(), groups);
+                groups.add(parent);
```

Comment:
Ahh!  intelliJ highlighted the old code as suspicious!  I love intelliJ!  Good catch!